src/moonlink/src/table_handler/test_utils.rs

```diff
@@ -221,13 +226,20 @@ impl TestEnvironment {
         self.send_event(TableEvent::StreamAbort { xact_id }).await;
     }
 
-    /// Force an index meger, and block wait its completion.
+    /// Force an index merge operation, and block wait its completion.
     pub async fn force_index_merge_and_sync(&self) {
         self.send_event(TableEvent::ForceIndexMerge).await;
         let mut index_merge_completion_rx = self.index_merge_completion_tx.subscribe();
         index_merge_completion_rx.recv().await.unwrap();
     }
 
+    /// Force a data compaction operation, and block wait its completion.
+    pub async fn force_data_compaction_and_sync(&self) {
```

Comment:
I actually considered using `TableEventManager` for command initiation and synchronization, the blocker I met is, `initial_xxx_command` takes a mutable reference which makes it hard to do concurrent testing...